Lib/pathlib/_types.py

```diff
@@ -0,0 +1,22 @@
+"""
+Protocols for supporting classes in pathlib.
+"""
+from typing import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class Parser(Protocol):
+    """Protocol for path parsers, which do low-level path manipulation.
+
+    Path parsers provide a subset of the os.path API, specifically those
+    functions needed to provide PurePathBase functionality. Each PurePathBase
+    subclass references its path parser via a 'parser' class attribute.
+    """
+
+    sep: str
```

Comment:
It's possible to drop this attribute. We can call `parser.join('a', 'a').strip('a')` to compute the separator from `PurePathBase.__init_subclass__()`, and then stash the result as `cls._sep`.

Reply:
I've opened this as a separate PR: #127725. I _think_ it's a good idea because it makes the interface method-only, and so allows `issubclass()` checks to work.